### ISSUE TITLE
Cannot find module 'exists-sync'

### ISSUE THREAD

**Issue by adewaguru** [awslabs/aws-cognito-angular2-quickstart#1]
I'm getting below error when I run "npm start". I'm running this in Windows 7 and only step that I didn't follow is creating the AWS resources by running the shell script.

---

C:\Users\ABCD\Desktop\Capital Cranes\aws-cognito-angular2-quickstart-master>npm start

> aws-cognito-angular2-quickstart@1.0.0 start C:\Users\ABCD\Desktop\Capital Cranes\aws-cognito-angular2-quickstart-master
> ng server

module.js:327
    throw err;
    ^

Error: Cannot find module 'exists-sync'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Users\ABCD\Desktop\Capital Cranes\aws-cognito-angular2-quickstart-master\node_modules\angular-cli\node_modules\ember-cli\lib\models\project.js:11:26)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\Program Files\nodejs\node.exe" "C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js" "start"
npm ERR! node v4.4.5
npm ERR! npm  v2.15.5
npm ERR! code ELIFECYCLE
npm ERR! aws-cognito-angular2-quickstart@1.0.0 start: `ng server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the aws-cognito-angular2-quickstart@1.0.0 start script 'ng server'.
npm ERR! This is most likely a problem with the aws-cognito-angular2-quickstart package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ng server
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs aws-cognito-angular2-quickstart
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls aws-cognito-angular2-quickstart
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\ABCD\Desktop\Capital Cranes\aws-cognito-angular2-quickstart-master\npm-debug.log

C:\Users\ABCD\Desktop\Capital Cranes\aws-cognito-angular2-quickstart-master>

**Comment by ahdach** [comment]
Hi @adewaguru ,

I've just cloned and launched the app and have the same error.
Did you solve it?

Here is my log (I'm running on mac) showing the same 'ng server' error.

12 error Darwin 14.1.0
13 error argv "node" "/usr/local/bin/npm" "start"
14 error node v0.12.0
15 error npm  v2.5.1
16 error code ELIFECYCLE
17 error aws-cognito-angular2-quickstart@1.0.0 start: `ng server`
17 error Exit status 1
18 error Failed at the aws-cognito-angular2-quickstart@1.0.0 start script 'ng server'.
18 error This is most likely a problem with the aws-cognito-angular2-quickstart package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     ng server
18 error You can get their info via:
18 error     npm owner ls aws-cognito-angular2-quickstart
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]

I will start digging now.

Best,
A

**Comment by ahdach** [comment]
@adewaguru and others,

Well.
It's a bit embarrassing as I made a very rookie mistake: not installing the requirements.

I'm new to angular, and, frankly, to development in general...

So, first I installed the requirements: aws cli, npm, bower and angular-cli.
Then, I noticed the error log changed. And was referring to missing modules.
I installed these modules. One by one. (Maybe I wouldn't have to do this if I had everything installed properly in the first place!).

And then, another npm start. And, voilà!, I can see the first page on the local host.

Hope this will help someone, someday.

So, @vbudilov , so far so good, your contrib rocks!
Thanks
A
